torch/_dynamo/variables/distributed.py

```diff
@@ -0,0 +1,113 @@
+import inspect
+from typing import Dict, List
+
+import torch
+from ..utils import istype
+from .base import VariableTracker
+
+
+def is_from_local(value):
+    if torch.distributed.is_available():
+        from torch.distributed._tensor import DTensor
+
+        if inspect.isfunction(value) and value is DTensor.from_local:
+            return True
+
+    return False
+
+
+class PlacementClassVariable(VariableTracker):
+    def __init__(self, value, **kwargs):
+        super().__init__(**kwargs)
+        self.value = value
```

Comment:
consider a class between these and `VariableTracker` - something like `DistributedVariableTracker` where init does `if not torch.distributed.is_available(): unimplemented(...)`